addons/account/models/account_reconcile_model.py

```diff
@@ -448,7 +448,10 @@ def _prepare_reconciliation(self, st_line, aml_ids=[], partner=None):
 
         reconciliation_overview, open_balance_vals = st_line._prepare_reconciliation(lines_vals_list)
 
-        writeoff_vals_list = self._get_write_off_move_lines_dict(st_line, write_off_amount)
+        residual_balance = (open_balance_vals and open_balance_vals['currency_id'] != self.company_id.currency_id.id) \
+                           and open_balance_vals['amount_currency'] or write_off_amount
+
+        writeoff_vals_list = self._get_write_off_move_lines_dict(st_line, residual_balance)
```

Comment:
unfortunately, this is far more complicated... All this method is expressed in company's currency. The root cause of this issue is there:
https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_reconcile_model.py#L368

Since currency_id is False, the computation are made in statement line currency. You should try to set the company's currency instead.
Btw: the code is different in master:
https://github.com/odoo/odoo/blob/master/addons/account/models/account_reconcile_model.py#L415

Reply:
I think the computation is made before, as in the writeoff_line we already set the balance in the company currency instead of the journal currency, it is never converted after that, and we call _get_write_off_move_lines_dict here with the wrong write_off_amount https://github.com/odoo/odoo/blob/cb8cff5a053999b6fde7496176fe6786b2c13933/addons/account/models/account_reconcile_model.py#L451